info_encoding.md

```diff
@@ -1,116 +1,108 @@
 ```
 (a, b, c)
 ```
 is encoded as:
 ```
-uint16BE(len(a)) || a || uint16BE(len(b)) || b || uint16BE(len(c)) || c
+encodeU16(len(a)) || a || encodeU16(len(b)) || b || encodeU16(len(c)) || c
+```
+
+## Ordered Key-Value Datasets
+
+Ordered key-value datasets can be encoded by first constructing and then encoding a list from the dataset.  To construct the list, start with an empty list, iterate over the dataset, and then for each key-value pair in the dataset first add the key and then the value to the list.
+
+### Example
+
+The list representation of the dataset
+
+```json
+{
+	"key1": "value1",
+	"key2": "value2"
+}
+```
+is given by
+```
+("key1", "value1", "key2", "value2")
 ```
 
 ## Nested Data
 
-Later applications may require nested data.
-Here, we propose two ways to achieve this feature in a backwards-compatible
-fashion.
-Again, we stress that due to the lack of schemalessness, new derivations
-can assign whatever meaning to the contents of the buffers.
-This does not limit the usefulness or security of the simpler, shallow
-encoding.
+Later applications may require nested data. Here, we propose two ways to achieve this feature in a backwards-compatible fashion. Again, we stress that due to the lack of schemalessness, new derivations can assign whatever meaning to the contents of the buffers. This does not limit the usefulness or security of the simpler, shallow encoding.
 
 ### Path-Style Keys
 
-When using the lists to describe key-value sets, the keys can be chosen
-to represent paths.
-This requires defining a path delimiter (e.g. '.', '/' or '\0'), or using
-length-prefixed lists to separate path segments.
-Then, if two paths begin with the same sequence of path segments, they
-are in the same compound data structure.
-The compound data structure itself is named by the shared prefix.
+When using the lists to describe key-value sets, the keys can be chosen to represent paths. This requires defining a path delimiter (e.g. '.', '/' or '\0'), or using length-prefixed lists to separate path segments. Then, if two paths begin with the same sequence of path segments, they are in the same compound data structure. The compound data structure itself is named by the shared prefix.
 
 ### _Recursive Length Prefix (RLP)_ Encoding
 
-RLP is similar to SLP, except that it allows list elements to themselves
-be lists.
-If that is the case, the element is first RLP-encoded itself, before the
-length is taken and it is concatenated.
+RLP is similar to SLP, except that it allows list elements to themselves be lists. If that is the case, the element is first RLP-encoded itself, before the length is taken and it is concatenated.
 
 Example: The list
 ```
 (a, (b, c))
 ```
 becomes:
 ```
-uint16BE(len(a)) || uint16BE(4 + len(b) + len(c)) || uint16BE(b) || b || uint16BE(c) || c
+encodeU16(len(a)) || a || encodeU16(4 + len(b) + len(c)) || encodeU16(b) || b || encodeU16(c) || c
 ```
 
-<a href name=Code>&nsbp;</a>
+<a name="section-code"> </a>
```

Comment:
oh, you don't need this for many markdowns (e.g. gh-markdown).
Titles get automatically converted into kebab-cased anchors.